### PR TITLE
Remove children's dob from event message context

### PR DESCRIPTION
**Pull request by YetAnotherMatt**
Children's DoB can now be obtained from the current IdentityAndEligibilityResponse, stored on the claim. We don't need to pass them in messages any more.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/payload/ReportPropertiesFactory.java [hunk 1]
```diff
@@ -95,11 +100,13 @@ protected Long calculateQueueTime(LocalDateTime timestamp) {
...
-        Claimant claimant = context.getClaim().getClaimant();
+        Claim claim = context.getClaim();
+        Claimant claimant = claim.getClaimant();
         LocalDate atDate = context.getTimestamp().toLocalDate();
         customMetrics.put(CLAIMANT_AGE.getFieldName(), Period.between(claimant.getDateOfBirth(), atDate).getYears());
-        long childrenUnder4 = getNumberOfChildrenUnderFour(context.getDatesOfBirthOfChildren(), context.getTimestamp().toLocalDate());
-        long childrenUnder1 = getNumberOfChildrenUnderOne(context.getDatesOfBirthOfChildren(), context.getTimestamp().toLocalDate());
+        List<LocalDate> dobOfChildrenUnder4 = claim.getCurrentIdentityAndEligibilityResponse().getDobOfChildrenUnder4();
+        long childrenUnder4 = getNumberOfChildrenUnderFour(dobOfChildrenUnder4, context.getTimestamp().toLocalDate());
+        long childrenUnder1 = getNumberOfChildrenUnderOne(dobOfChildrenUnder4, context.getTimestamp().toLocalDate());
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/payload/ReportPropertiesFactory.java]: You're calling `context.getTimestamp().toLocalDate()` in both these lines, that could be refactored out.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantLifecycleIntegrationTests.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ void shouldProcessClaimFromPregnancyToChildTurningFour() throws JsonProcessingEx
 
         UUID claimId = applyForHealthyStartAsPregnantWomanWithNoChildren(expectedDeliveryDate);
         Claim claim = repositoryMediator.loadClaim(claimId);
-        wiremockManager.verifyGoogleAnalyticsCalledForClaimEventWithNoChildren(claim, UPDATED_FROM_NEW_TO_ACTIVE);
+        wiremockManager.verifyGoogleAnalyticsCalledForClaimEvent(claim, UPDATED_FROM_NEW_TO_ACTIVE);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantLifecycleIntegrationTests.java]: Why has this changed?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/PaymentCycleIntegrationTests.java [hunk 3]
```diff
@@ -285,18 +286,20 @@ void shouldSendNoLongerEligibleEmailWhenEligibleWithNoChildrenOnFeedAndNotPregna
         wiremockManager.stubSuccessfulEligibilityResponse(currentPaymentCycleChildrenDobs);
         stubNotificationEmailResponse();
 
-        Claim claim = createActiveClaimWithPaymentCycleEndingYesterday(previousPaymentCycleChildrenDobs, NOT_PREGNANT);
+        UUID claimId = createActiveClaimWithPaymentCycleEndingYesterday(previousPaymentCycleChildrenDobs, NOT_PREGNANT).getId();
 
         invokeAllSchedulers();
 
+        Claim claim = repositoryMediator.loadClaim(claimId);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/PaymentCycleIntegrationTests.java]: Why did you need to do this?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/payload/ReportPropertiesFactory.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/payload/ReportPropertiesFactory.java]: true, though that was already the case

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantLifecycleIntegrationTests.java (hunk 2 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantLifecycleIntegrationTests.java]: Now that we're obtaining the list of children's dob from the claim, there's no need to pass a list through to wiremock - which means that there's no need to a separate `verifyGoogleAnalyticsCalledForClaimEventWithNoChildren` method.
